nx/lib/nx/lin_alg/svd.ex

```diff
@@ -200,12 +202,17 @@ defmodule Nx.LinAlg.SVD do
         {u, l, iter_idx + 1, is_unconverged, is_not_max_iteration, max_iter}
       end
 
+    u = Nx.as_type(u, original_type)
     u = 1.5 * u - 0.5 * Nx.dot(u, Nx.dot(Nx.LinAlg.adjoint(u), u))
     h = u |> Nx.LinAlg.adjoint() |> Nx.dot(x)
     h = (h + Nx.LinAlg.adjoint(h)) / 2
     {u, h}
   end
 
+  # f16 is not enough precision to compute SVD
+  deftransformp min_precision_type({:f, 64}), do: {:f, 64}
```

Comment:
IIRC the code only works for f32 (even Jax enforces this), but I might be wrong.